src/evaluation/Run.jl

```diff
@@ -54,14 +55,9 @@ function run_reactive!(session::ServerSession, notebook::Notebook, old_topology:
     to_run_raw = setdiff(union(new_runnable, old_runnable), keys(new_order.errable))::Vector{Cell} # TODO: think if old error cell order matters
 
     # find (indirectly) deactivated cells and update their status
-    deactivated = filter(c -> c.running_disabled, notebook.cells)
-    indirectly_deactivated = collect(topological_order(notebook, new_topology, deactivated))
-    for cell in indirectly_deactivated
-        cell.running = false
-        cell.queued = false
-        cell.depends_on_disabled_cells = true
-    end
-
+	if indirectly_deactivated === nothing
+		indirectly_deactivated = disable_dependent_cells!(notebook, new_topology)
```

Comment:
This may be a piece of premature optimization on my side ;-) 
I'll try your last suggestion to just re-call `disable_dependent_cells!`. If it turnes out to be a performance bottleneck we can still optimize it later.